### DISCUSSION TITLE
GND calculation issue

### DISCUSSION THREAD

**Discussion by catalin-pr** [mtex-toolbox/mtex#524] [Ask Anything] [answered]
ear All, 

I am trying to calculate the GND for an.osc file however during the loading and running the code I face some problems. 
The following warning I face when I load the data :
1. "Warning: more column data was passed in than expected. Check your column names make sense!"
2. "Warning: You have choosen to correct your EBSD data for differently aligned reference
frames for the Euler angles and the map coordinates. However, you have not specified which
reference system setting has been used on your Edax system . I'm going to assume "setting
1". Be careful, the default setting of EDAX is "setting 2". Click here for more
information.Please make sure you have chosen the correct setting and specify it
explicitely using the syntax

 ebsd = EBSD.load(fileName,'convertSpatial2EulerReferenceFrame','setting 2')
> In loadEBSD_osc (line 90)
  In EBSD.load (line 86)

Grid size (hex): 986 x 1408"

and then during the calculator of curvature I get:

"No appropriate method, property, or field 'curvature' for class 'EBSDhex'.

Error in EBSD/subsref (line 40)
  [varargout{1:nargout}] = builtin('subsref',ebsd,s);"
 
I want to inform taht I use MTEX 5.2.7.

Much appreciated your help.

Many regards,
Catalin

**Comment by ralfHielscher** [comment]
Hi Catalin

1. you can ignore this warning.
2. The warning comes with a link with more information. You should have clicked it :smile: https://mtex-toolbox.github.io/EBSDReferenceFrame.html . 
3. This is indeed a problem. If it is very important for you I can do it on the weekend.

Ralf.

**Comment by catalin-pr** [comment]
Hi Ralf,

Indeed, I ignored the warning and checked point 2 ...however I am stuck
with point 3.

If you have some spare time, may I can share the file and you can look over
the weekend.

I even tried to change the phase, form alpha to beta to see if I can obtain
any further info and was nothing good.

Many thanks,
Catalin

**Reply by catalin-pr** [reply]
Hi Ralf,

I  am wonder if by any chance do you had a look to see which could be the problem with ""No appropriate method, property, or field 'curvature' for class 'EBSDhex'." 

MAny thanks,
Catalin

**Comment by ralfHielscher** [comment, marked as answer]
Hi Catalin,

actually I did it already :smile: If you download the development branch it should already be working. I just put a new beta online.

Ralf.

**Reply by catalin-pr** [reply]
Hi Ralf, 

Many thanks indeed. Do you refer this issue is solved in MTEX 5.5.beta.3 ?

Thanks a lot,
Catalin

**Comment by ralfHielscher** [comment]
look here https://github.com/mtex-toolbox/mtex/releases
I have not yet updated the links.

Ralf.

**Reply by catalin-pr** [reply]
I used 5.5.4 and generate another error:

Output argument "gX" (and maybe others) not assigned during call to "EBSDhex/get.gradientX".

Error in EBSD/curvature (line 16)
kappa = dyad(ebsd.gradientX,vector3d.X) + ...

Error in EBSD/subsref (line 40)
  [varargout{1:nargout}] = builtin('subsref',ebsd,s);

Many thanks,
Catalin

**Reply by ralfHielscher** [reply]
I uploaded a next beta 5.5.beta.5. Could you give it a try.

Ralf.

**Reply by catalin-pr** [reply]
Ok, I will try and next inform you the outcome.
Indeed the new version is faster..

Thanks
Catalin

**Comment by catalin-pr** [comment]
Hi Ralf, 

Thank a lot for your help, indeed this work.

Many regards,
Catalin
